File: exercise/C#/day11/Christmas.Tests/PreparationTests.cs
using FluentAssertions;
using Xunit;
using static Christmas.ToyType;

namespace Christmas.Tests
{
    public class PreparationTests
    {
        [Theory]
        [InlineData(-1, "No gifts to prepare.")]
        [InlineData(0, "No gifts to prepare.")]
        [InlineData(1, "Elves will prepare the gifts.")]
        [InlineData(49, "Elves will prepare the gifts.")]
        [InlineData(50, "Santa will prepare the gifts.")]
        public void PrepareGifts(int numberOfGifts, string expected)
            => Preparation.PrepareGifts(numberOfGifts);

        [Theory]
        [InlineData(1, "Baby")]
        [InlineData(3, "Toddler")]
        [InlineData(6, "Child")]
        [InlineData(13, "Teen")]
        public void CategorizeGift(int age, string expectedCategory)
            => Preparation.CategorizeGift(age)
                .Should()
                .Be(expectedCategory);

        [Theory]
        [InlineData(Educational, 25, 100, true)]
        [InlineData(Fun, 30, 100, true)]
        [InlineData(Creative, 20, 100, true)]
        [InlineData(Educational, 20, 100, false)]
        [InlineData(Fun, 29, 100, false)]
        [InlineData(Creative, 15, 100, false)]
        public void EnsureToyBalance(ToyType toyType, int toysCount, int totalToys, bool expected)
            => Preparation.EnsureToyBalance(toyType, toysCount, totalToys)
                .Should()
                .Be(expected);
    }
}

File: exercise/java/day11/src/test/java/PreparationTests.java
import christmas.Preparation;
import christmas.ToyType;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.CsvSource;

import static org.assertj.core.api.Assertions.assertThat;

class PreparationTests {

    @ParameterizedTest
    @CsvSource({
            "-1, No gifts to prepare.",
            "0, No gifts to prepare.",
            "1, Elves will prepare the gifts.",
            "49, Elves will prepare the gifts.",
            "50, Santa will prepare the gifts."
    })
    void prepareGifts(int numberOfGifts, String expected) {
        String result = Preparation.prepareGifts(numberOfGifts);
        assertThat(result).isEqualTo(expected);
    }

    @ParameterizedTest
    @CsvSource({
            "1, Baby",
            "3, Toddler",
            "6, Child",
            "13, Teen"
    })
    void categorizeGift(int age, String expectedCategory) {
        String result = Preparation.categorizeGift(age);
        assertThat(result).isEqualTo(expectedCategory);
    }

    @ParameterizedTest
    @CsvSource({
            "EDUCATIONAL, 25, 100, true",
            "FUN, 30, 100, true",
            "CREATIVE, 20, 100, true",
            "EDUCATIONAL, 20, 100, false",
            "FUN, 29, 100, false",
            "CREATIVE, 15, 100, false"
    })
    void ensureToyBalance(ToyType toyType, int toysCount, int totalToys, boolean expected) {
        boolean result = Preparation.ensureToyBalance(toyType, toysCount, totalToys);
        assertThat(result).isEqualTo(expected);
    }
}
